src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/SourceMonitorParser.java

```diff
@@ -91,20 +93,36 @@ public SourceMonitorReport invoke(java.io.File workspace, VirtualChannel channel
 		    metricNameMap.put(metricNameElt.getAttributeValue("id"), metricNameElt.getValue());
         }
 
+
+        ArrayList<FunctionStats> detailedMetrics = new ArrayList<FunctionStats>();
+
 		// Parse Summary checkpoint data.
         Element checkpoints = projectElt.getChild("checkpoints");
         List<?> checkpointsEltList = checkpoints.getChildren();
         for (int i = 0; i < checkpointsEltList.size(); i++) {
             Element checkpoint = (Element)checkpointsEltList.get(i);
             Element metricsElt = checkpoint.getChild("metrics");
             List<?> metricsEltList = metricsElt.getChildren();
+            Element functionMetrics = checkpoint.getChild("function_metrics");
+            List<?> functionMetricsEltList = functionMetrics.getChildren("function");
+
             for (int j = 0; j < metricsEltList.size(); j++) {
                 Element metricElt = (Element)metricsEltList.get(j);
                 metricsSummaryMap.put(metricNameMap.get(metricElt.getAttributeValue("id")), metricElt.getValue());
             }
+            int numFunctions = Integer.parseInt(functionMetrics.getAttributeValue("function_count"));
+            for (int k = 0; k < numFunctions; k++){

```

Comment:
The function is getting pretty complex. Maybe break out the new code into a separate function.